### PR TITLE
tests.yml: Deploy the Docker image to Docker Hub

### PR DESCRIPTION
**Pull request by sjackman**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

After this PR, we can stop building Docker images using Docker Hub. The Docker image will be built and deployed using GitHub Actions.

Please do not merge this PR until after the next tag of Homebrew/brew.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -174,30 +174,27 @@ jobs:
...
-        docker pull homebrew/brew
-        docker-compose -f Dockerfile.yml build sut
+      run: docker build -t brew .
 
     - name: Run brew test-bot
       run: |
         if [ "$RUNNER_OS" = "Linux" ]; then
-          docker-compose -f Dockerfile.yml run --rm -v $GITHUB_WORKSPACE:/tmp/test-bot sut
-          docker tag homebrew_sut brew
+          docker run --rm -v $GITHUB_WORKSPACE:/tmp/test-bot brew brew test-bot
```
Comment (sjackman) [inline .github/workflows/tests.yml:182]: @MikeMcQuaid @reitermarkus Why is this `-v` volume mount option here? Does `brew test-bot` produce any build artifacts used later? It doesn't look like it.

Comment (MikeMcQuaid) [reply .github/workflows/tests.yml:182]: Probably just copied from a place where the artefacts were used later.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -174,30 +174,27 @@ jobs:
 
     - name: Build Docker image
       if: matrix.os == 'ubuntu-latest'
-      run: |
-        docker pull homebrew/brew
-        docker-compose -f Dockerfile.yml build sut
+      run: docker build -t brew .
```
Comment (MikeMcQuaid) [inline .github/workflows/tests.yml:177]: Is there any caching we can do with this somehow?

Comment (MikeMcQuaid) [reply .github/workflows/tests.yml:177]: e.g. https://github.com/marketplace/actions/build-docker-images-using-cache

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -174,30 +174,27 @@ jobs:
...
-    - name: Deploy the latest Docker image to GitHub
-      if: matrix.os == 'ubuntu-latest' && github.ref == 'refs/heads/master'
-      run: |
-        docker login docker.pkg.github.com -u BrewTestBot -p ${{secrets.GITHUB_TOKEN}}
-        docker tag brew docker.pkg.github.com/homebrew/brew/brew
-        docker push docker.pkg.github.com/homebrew/brew/brew
-
-    - name: Deploy the tagged Docker image to GitHub
-      if: matrix.os == 'ubuntu-latest' && startsWith(github.ref, 'refs/tags/')
+    - name: Deploy the Docker image to GitHub and Docker Hub
```
Comment (MikeMcQuaid) [inline .github/workflows/tests.yml:187]: How long does this take?

File: .github/workflows/tests.yml (hunk 1 above)
Comment (sjackman) [reply .github/workflows/tests.yml:182]: I've removed it.

File: .github/workflows/tests.yml (hunk 3 above)
Comment (sjackman) [reply .github/workflows/tests.yml:187]: 33 seconds. See https://github.com/Homebrew/brew/runs/643203256
Deployment doesn't happen in the PR check though, only for `master` and tags.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (sjackman) [reply .github/workflows/tests.yml:177]: Yes, but I don't believe it's worth the additional complexity. On my machine, the cachable portion takes 50s to run, and the non-cacheable portion takes 3m47s to run, and the whole CI run takes 18m55s to run, so there's not a great saving to be had by caching the base layer.

Comment (sjackman) [reply .github/workflows/tests.yml:177]: The `Build Docker image` step takes 3m26s on GitHub Actions.

File: .github/workflows/tests.yml (hunk 3 above)
Comment (MikeMcQuaid) [reply .github/workflows/tests.yml:187]: Cool, sounds good, thanks 👍

File: .github/workflows/tests.yml (hunk 2 above)
Comment (MikeMcQuaid) [reply .github/workflows/tests.yml:177]: Yeh, fair enough.

**Comment by MikeMcQuaid** [comment]
> Please do not merge this PR until after the next tag of Homebrew/brew.

Tagged, merged!

**Pull request merged**